parsers/US_SPP.py

```diff
@@ -52,7 +55,22 @@ def data_processor(df, logger):
     # Remove leading whitespace in column headers.
     df.columns = df.columns.str.strip()
 
-    keys_to_remove = {'GMT MKT Interval', 'Average Actual Load', 'Load', 'Other', 'Waste Heat'}
+    #Some historical csvs split the production into 'Market' and 'Self',
+    #So first we need to combine those.
+    for col in df.columns:
+        if 'Market' in col:
+            combined_col = col.replace('Market','').strip()
+            self_col = col.replace('Market','Self')
+            if self_col in df.columns:
+                df[combined_col] = df[col] + df[self_col]
+                df.drop(self_col, inplace=True, axis=1)
+            else:
+                logger.warning(f'Corresponding column "{self_col}" to "{col}" not found in file', extra={'key':'US-SPP'})
```

Comment:
I think the new log message is causing https://github.com/tmrowco/electricitymap-contrib/blob/master/parsers/test/test_US_SPP.py#L63 to break.

Reply:
Hey systemcatch! That is strange indeed. Are you referring to when running "python test_parser.py US-SPP", because I do not get an error when running those tests. Could you give me some more details on how you would suggest fixing it? Thanks for checking on this!

Reply:
I understand now! Nevermind, I am looking into it, you are probably right that the logger would cause an issue.